src/net/sourceforge/kolmafia/FamiliarData.java

```diff
@@ -617,6 +619,67 @@ public boolean getFeasted() {
     return this.feasted;
   }
 
+  public void deactivate() {
+    // Do anything necessary when this familiar is banished to the Terrarium
+    this.active = false;
+    switch (this.id) {
+      case FamiliarPool.GREY_GOOSE:
+        removeGreyGooseSkills();
+        break;
+    }
+  }
+
+  public void activate() {
+    // Do anything necessary when this familiar is removed from the Terrarium
+    this.active = true;
+    switch (this.id) {
+      case FamiliarPool.GREY_GOOSE:
+        if (this.weight >= 6) {
+          addGreyGooseSkills();
+        }
+        break;
+    }
+  }
+
+  private void changeWeight(int oldWeight, int newWeight) {
+    if (oldWeight == newWeight) {
+      return;
+    }
+    switch (this.id) {
+      case FamiliarPool.GREY_GOOSE:
+        if (this.active) {
+          if (oldWeight < 6 && newWeight >= 6) {
```

Comment:
`addAvailableCombatSkill` and the `remove` version seem idempotent to me, so you could do without the `oldWeight` checks. I'm not sure whether, overall, that's better or worse.